### PR TITLE
Bump Celery to 4.2.0

### LINKED ISSUE #5221: Conflicting versions of celery and kombu in requirements.txt
Make sure these boxes are checked before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if any
- [X] I have reproduced the issue with at least the latest released version of superset
- [X] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
Master branch

### Expected results
Pipping from `requirements.txt` should work

### Actual results
pip installation fails with `celery 4.1.1 has requirement kombu<5.0,>=4.2.0, but you'll have kombu 4.1.0 which is incompatible.`

### Steps to reproduce
`pip install -r requirements.txt`

### PR DESCRIPTION
**Pull request by villebro**
More background can be found in https://github.com/celery/kombu/issues/870, `Celery` needs to be fixed at `4.1.0` until `Kombu 4.2.2` arrives. Fixes #5221.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ geopy==1.11.0
 gunicorn==19.8.0
 humanize==0.5.1
 idna==2.6
-kombu==4.1.0
+kombu<4.2
```
Comment (john-bodley) [inline requirements.txt]: Can we just remove this line altogether? Superset doesn't directly use `kombo` it and it was added in part to circumvent the issue. Also packages in `requirements.txt` should be pinned to a specific version.

cc: @timifasubaa

**Comment by john-bodley** [comment]
@villebro so for iterating on the comment. When I tested this locally (to see if we still needed to pin `kombu` when using `celery==4.1.0`),
```
> pyenv version
2.7.13 (set by /Users/john_bodley/.pyenv/version)
> virtualenv env
> pip install celery==4.1.0
...
Collecting kombu<5.0,>=4.0.2 (from celery==4.1.0)
...
> pip freeze | grep kombu
kombu==4.2.1
```
It seems like the version of `kombu` might still be problematic given the range versioning defined from `celery==4.1.0`.

**Comment by villebro** [comment]
Oops, good call @john-bodley. I repinned `Kombu` at `4.1.0`.

**Review by john-bodley — APPROVED** [review]

**Comment by john-bodley** [comment]
Actually @villebro they've recently released `celery` version [4.2.0](https://pypi.org/project/celery/4.2.0/) and I was wondering whether bumping the version higher would remedy the issue. If so we should pin `celery==4.2.0` in `requirements.txt` and require `celery>=4.2.0` in `setup.py` to ensure we bypass this issue.

**Comment by villebro** [comment]
Let's see what happens..

**Comment by villebro** [comment]
Looking at CI logs bumping to 4.2.0 seems to have resolved the issue!

**Review by john-bodley — APPROVED** [review]

**Comment by john-bodley** [comment]
Thanks @villebro.

**Comment by john-bodley** [comment]
For reference here’s the item in Celery 4.2.0 [changelog](https://github.com/celery/celery/blob/master/Changelog#L346) which remedies the issue.

**Pull request merged**